package/windows/cleanup.ps1

```diff
@@ -1,144 +1,150 @@
-if (-not $currentPrincipal.IsInRole([System.Security.Principal.WindowsBuiltInRole]::Administrator)) {
-    throw "You need elevated Administrator privileges in order to run this script, start Windows PowerShell by using the Run as Administrator option"
+# remove rancher-wins service
+Log-Info "Stopping rancher-wins service ..."
+Get-Service -Name "rancher-wins" -ErrorAction Ignore | Where-Object {$_.Status -eq "Running"} | Stop-Service -Force -ErrorAction Ignore
+Push-Location c:\etc\rancher
+$errMsg = $(.\wins.exe srv app run --unregister)
+if (-not $?) {
+    Log-Warn "Could not unregister rancher-wins service: $errMsg"
+}
+Pop-Location
+
+# stop kubernetes components processes
+Get-Process -ErrorAction Ignore -Name @(
+    "flanneld*"
+    "kubelet*"
+    "kube-proxy*"
+    "nginx*"
+    "wins"
+    "rancher-wins-*"
+) | ForEach-Object {
+    Log-Info "Stopping $( $_.Name ) process ..."
+    $_ | Stop-Process -Force -ErrorAction Ignore
 }
 
-try {
-    $svcAgent = Get-Service -Name "rancher-agent" -ErrorAction Ignore
-    if ($svcAgent) {
-        if ($svcAgent.Status -eq "Running") {
-            $svcAgent | Stop-Service -Force -ErrorAction Ignore
-        }
-
-        pushd $RancherDir
-        .\agent.exe --unregister-service *>$null
-        popd
-    }
-} catch { }
-
-try {
-    # flanneld #
-    $process = Get-Process -Name "flanneld*" -ErrorAction Ignore
-    if ($process) {
-        $process | Stop-Process -Force | Out-Null
-    }
-
-    # kubelet #
-    $process = Get-Process -Name "kubelet*" -ErrorAction Ignore
-    if ($process) {
-        $process | Stop-Process -Force | Out-Null
-    }
-
-    # kube-proxy #
-    $process = Get-Process -Name "kube-proxy*" -ErrorAction Ignore
-    if ($process) {
-        $process | Stop-Process -Force | Out-Null
-    }
-
-    # controlplanes proxy #
-    $process = Get-Process -Name "nginx*" -ErrorAction Ignore
-    if ($process) {
-        $process | Stop-Process -Force | Out-Null
-    }
-
-    # docker stop #
-    docker ps -q | % { docker stop $_ *>$null } *>$null
+# clean up docker conatiner: docker rm -fv $(docker ps -qa)
+Log-Info "Removing Docker containers ..."
+$errMsg = $(docker.exe ps -aq | ForEach-Object {docker.exe rm -f $_})
+if (-not $?) {
+    Log-Warn "Could not remove docker containers: $errMsg"
+}
 
-    # clean up rancher parts #
-    docker rm kubernetes-binaries *>$null
-    docker rm cni-binaries *>$null
-} catch { }
+# clean network interface
+Log-Info "Removing network settings ..."
+Get-HnsNetwork | Where-Object {@('cbr0', 'vxlan0') -contains $_.Name} | Remove-HnsNetwork
+Invoke-HNSRequest -Type policylists -Method GET | ForEach-Object {
+    Invoke-HNSRequest -Method DELETE -Type policylists -Id $_.Id
+}
 
-try {
-    $networkName = get-env-var "KUBE_NETWORK"
-    if ($networkName) {
-        $null = Clean-HNSNetworks -Names @{
-            $networkName = $True
-        }
-    }
+# clean firewall rules
+Log-Info "Removing firewall rules ..."
```

Comment:
Please use `Cleaning up` instead of `Removing`. You need to indicate the specific rule you are cleaning up in your log message.

Reply:
Modified